### PR TITLE
`azurerm_linux_web_app`, `azurerm_linux_web_app_slot` - fix docker runtime issue

### PR DESCRIPTION
**Pull request by xiaxyi**
Fixing `linuxFxVersion` issue. 

Adding docker registry url, username, password in site_config for linux web app, just like linux function does.
Supporting docker compose feature

Fix https://github.com/hashicorp/terraform-provider-azurerm/issues/21504
Fix https://github.com/hashicorp/terraform-provider-azurerm/issues/16290

```
--- PASS: TestAccLinuxWebAppSlot_withDockerCompose (257.49s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    258.206s

--- PASS: TestAccLinuxWebApp_withDockerCompose (247.97s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    248.728s


```

### CODE REVIEW

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @xiaxyi - thanks for this PR.
I started to review, but found that the tests do not currently pass for the new schema, could you take a look?
```
=== CONT  TestAccLinuxWebApp_withDocker
    testcase.go:110: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # azurerm_linux_web_app.test will be updated in-place
          ~ resource "azurerm_linux_web_app" "test" {
                id                                = "/subscriptions/*****/resourceGroups/acctestRG-230330131731620719/providers/Microsoft.Web/sites/acctestWA-230330131731620719"
                name                              = "acctestWA-230330131731620719"
                # (18 unchanged attributes hidden)
        
              ~ site_config {
                    # (22 unchanged attributes hidden)
        
                  ~ application_stack {
                      ~ docker_image     = "appsvc/staticsite" -> "mcr.microsoft.com/appsvc/staticsite"
                      - registry_url     = "https://mcr.microsoft.com" -> null
                        # (1 unchanged attribute hidden)
                    }
                }
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccLinuxWebApp_withDocker (507.76s)
```

File: internal/services/appservice/helpers/app_stack.go [hunk 1]
```diff
@@ -593,6 +621,35 @@ func linuxApplicationStackSchema() *pluginsdk.Schema {
...
+
+	if !features.FourPointOhBeta() {
+		s.Elem.(*pluginsdk.Resource).Schema["registry_url"] = &pluginsdk.Schema{
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "The URL of the docker registry.",
+		}
+
+		s.Elem.(*pluginsdk.Resource).Schema["docker_image"] = &pluginsdk.Schema{
```
[the remark this answers is not in the document]
Comment (xiaxyi) [reply internal/services/appservice/helpers/app_stack.go:633]: Thanks @jackofallops , we are not deprecating this property, just changing the property restrictions in 4.0 provider. `docker_image` needs to be used together with `regisrty_url`

File: internal/services/appservice/helpers/app_stack.go [hunk 2]
```diff
@@ -593,6 +621,35 @@ func linuxApplicationStackSchema() *pluginsdk.Schema {
 			},
 		},
 	}
+
+	if !features.FourPointOhBeta() {
+		s.Elem.(*pluginsdk.Resource).Schema["registry_url"] = &pluginsdk.Schema{
```
[the remark this answers is not in the document]
Comment (xiaxyi) [reply internal/services/appservice/helpers/app_stack.go:626]: as commented below

**Comment by xiaxyi** [comment]
Thanks @jackofallops for the review, updated the code.

**Comment by xiaxyi** [comment]
Thanks @jackofallops for the review, I updated the test cases and made a tiny change to the read function.

would like to explain why I added the import ignore to:
```
1. registry_username
2. registry_password
3. registry_url
```

For `registry_url`, this is not required in `app_stack` in 3.x provider, which means user can set it in `app_setting` and this may cause diff for current users. The situation is the same for the other twos.

**Comment by jackofallops** [comment]
Hi @xiaxyi - Looks like we still have test failures relating to this change, could you take a look?
```
=== CONT  TestAccLinuxWebAppSlot_withDocker
testcase.go:110: Step 1/2 error: Check failed: Check 2/2 error: azurerm_linux_web_app_slot.test: Attribute 'site_config.0.linux_fx_version' expected "DOCKER|mcr.microsoft.com/appsvc/staticsite:latest", got ""
--- FAIL: TestAccLinuxWebAppSlot_withDocker (229.00s)
```

**Comment by xiaxyi** [comment]
Thanks @jackofallops for the comment.

updated:
```
--- PASS: TestAccLinuxWebAppSlot_withDocker (601.20s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    601.904s


```

**Comment by jackofallops** [comment]
This PR has been superseded by 22003, as such I'm going to close this one out.

**Pull request closed without merging**